### PR TITLE
Implement authentication using nextauth and mongodb

### PR DESCRIPTION
**Pull request by Do-raa**
In this PR I worked on adding the sign in and sign up functions and forms to the application with nextauth and mongodb

### CODE REVIEW

File: app/lib/utils.ts
Comment (mhsmustafa84) [inline app/lib/utils.ts:1]: Good job

File: app/lib/auth.ts
Comment (mhsmustafa84) [inline app/lib/auth.ts:1]: Bravooooo

**Review by mhsmustafa84 — APPROVED** [review]
Great work

File: app/lib/utils.ts
Comment (Do-raa) [reply app/lib/utils.ts:1]: thank you :)

**Pull request merged**